src/Main.hs

```diff
@@ -119,7 +118,7 @@ closestIntersection :: Ray -> [Object] -> Maybe (Double, Object)
 closestIntersection ray objects
     | null intersections = Nothing
     | otherwise = Just $ minimumBy minimumDefinedByFirst intersections
-    where intersections = catMaybes $ map (intersects ray) objects
+    where intersections = mapMaybe (intersects ray) objects
```

Comment:
That function is a great find. To be clear: if intersects returns Nothing then it's filtered out. Otherwise it returns an object?